### ISSUE TITLE
World model rollouts are blurry; any recommended hyperparameters for training?

### ISSUE THREAD

**Issue by chrisgao99** [nicklashansen/dreamer4#9]
Hi, thank you for open-sourcing this project. I have been trying to reproduce/train the world model pipeline, but my dynamics rollouts are still quite blurry even after 150k training steps, and I wanted to ask whether you have recommended hyperparameters or debugging tips.

Here is the world model rollouts at 145k steps, and the loss total is converging below 0.003 after 60k steps
<img width="3088" height="1024" alt="Image" src="https://github.com/user-attachments/assets/902be1c7-f1c2-44b2-a506-7382c3031c22" />

My tokenizer looks fine. Here is my tokenizer rollouts at training steps 400k when the loss total is smaller than 0.005
<img width="1024" height="2048" alt="Image" src="https://github.com/user-attachments/assets/aef029f2-b935-4467-bc7c-a5715abf1cb5" />

For training dynamics, I used mostly the default configs and this is my command
```
torchrun --nproc_per_node=4 train_dynamics.py --use_actions --data_dirs ./data/expert ./data/mixed-small ./data/mixed-large --frame_dirs ./data/expert-shards ./data/mixed-small-shards ./data/mixed-large-shards --tasks_json ../tasks.json --seq_len 32 --batch_size 24 --grad_accum 1 --eval_every 5000 --save_every 10000 --log_every 200 --wandb_run_name dynamics_tok0415000_4xa100
```

To be more similar to the paper, maybe i should use the config like this?
```
torchrun --nproc_per_node=8 train_dynamics.py \
  --use_actions \
  --k_max 64 \
  --self_fraction 0.857142857 \
  --bootstrap_start 0 \
  --time_every 4
```

Do you suggest it? Or did you use another method I may have overlooked to improve the world model training? I would appreciate any help.

**Comment by StefanoBraghetto** [comment]
I am also interested to know more about the HP. I am actually much worse than your pictures @chrisgao99 
How many steps you did to get your results?

I know you can get most of the HP information from the tokenizer.py they shared (in case is useful).

Something that is interesting to me is that they don't seem to be using scheduler. Which is something that could improve results.
